### PR TITLE
populate missing callable prototypes

### PR DESCRIPTION
**Pull request by dktapps**
The missing prototype on the `@var` tag inside `createFilePhpDocMap()` had to be spotted by eye, but the others were detected by #120.

It might be a good idea to have a rule checking for missing generics & prototypes in @var.

### CODE REVIEW

File: src/Type/RecursionGuard.php [hunk 1]
```diff
@@ -8,6 +8,12 @@ class RecursionGuard
 	/** @var true[] */
 	private static $context = [];
 
+	/**
+	 * @param Type $type
+	 * @param callable(): Type $callback
```
Comment (staabm) [inline src/Type/RecursionGuard.php]: Is there a space too much?

```suggestion
	 * @param callable():Type $callback
```

Comment (dktapps) [reply src/Type/RecursionGuard.php]: No, there's only 1 space. This is consistent with everywhere else in the project.

Comment (staabm) [reply src/Type/RecursionGuard.php]: I see. I didnt  know that phpdocs are allowed to contain spaces

**Comment by ondrejmirtes** [comment]
Thank you! This approach of separating actually unrelated changes is great :)

**Comment by dktapps** [comment]
Picked that up from experience :)

**Pull request merged**